extensions/oidc/runtime/src/main/java/io/quarkus/oidc/Oidc.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.oidc;
+
+import io.quarkus.oidc.common.runtime.config.OidcClientCommonConfig;
+import io.quarkus.oidc.runtime.OidcTenantConfig.ApplicationType;
+
+/**
+ * A CDI event that facilitates programmatic OIDC extension setup.
+ * Static OIDC tenants must be created in a synchronous manner from an observer method like in the example below:
```

Comment:
I'd like to propose not to highlight a rather technical `static tenant` term, so the follow up suggestions are all related to it:

```suggestion
 * OIDC providers must be created from an observer method like in the example below:
```